client/network/src/service.rs

```diff
@@ -733,25 +733,116 @@ pub struct NetworkWorker<B: BlockT + 'static, H: ExHashT> {
 	/// Senders for events that happen on the network.
 	event_streams: Vec<mpsc::UnboundedSender<Event>>,
 	/// Prometheus network metrics.
-	metrics: Option<Metrics>
+	metrics: Option<Metrics>,
 }
 
 struct Metrics {
+	// This list is ordered alphabetically
+	average_download_per_sec: Gauge<U64>,
+	average_upload_per_sec: Gauge<U64>,
+	import_queue_block_submits: Counter<U64>,
+	import_queue_finality_proofs_submits: Counter<U64>,
+	import_queue_justification_submits: Counter<U64>,
 	is_major_syncing: Gauge<U64>,
+	kbuckets_num_nodes: Gauge<U64>,
+	libp2p_connections: Gauge<U64>,
+	notifications_in: CounterVec<U64>,
+	notifications_out: CounterVec<U64>,
+	num_event_stream_channels: Gauge<U64>,
+	opened_notification_streams: GaugeVec<U64>,
 	peers_count: Gauge<U64>,
+	peerset_num_discovered: Gauge<U64>,
+	peerset_num_requested: Gauge<U64>,
+	random_kademalia_queries: Counter<U64>,
 }
 
 impl Metrics {
 	fn register(registry: &Registry) -> Result<Self, PrometheusError> {
 		Ok(Self {
+			// This list is ordered alphabetically
+			average_download_per_sec: register(Gauge::new(
+				"sub_libp2p_average_download_per_sec", "Average number of bytes downloaded per second",
+			)?, registry)?,
+			average_upload_per_sec: register(Gauge::new(
+				"sub_libp2p_average_upload_per_sec", "Average number of bytes uploaded per second",
+			)?, registry)?,
+			import_queue_block_submits: register(Counter::new(
+				"import_queue_block_submits",
+				"Number of blocks submitted to the import queue.",
+			)?, registry)?,
+			import_queue_finality_proofs_submits: register(Counter::new(
+				"import_queue_finality_proofs_submits",
+				"Number of finality proofs submitted to the import queue.",
+			)?, registry)?,
+			import_queue_justification_submits: register(Counter::new(
+				"import_queue_justification_submits",
+				"Number of justifications submitted to the import queue.",
+			)?, registry)?,
 			is_major_syncing: register(Gauge::new(
-				"is_major_syncing", "Whether the node is performing a major sync or not.",
+				"sub_libp2p_is_major_syncing", "Whether the node is performing a major sync or not.",
+			)?, registry)?,
+			libp2p_connections: register(Gauge::new(
+				"sub_libp2p_libp2p_connections", "Number of libp2p connections to a given node"
+			)?, registry)?,
+			kbuckets_num_nodes: register(Gauge::new(
+				"sub_libp2p_kbuckets_num_nodes", "Number of nodes in the Kademlia k-buckets"
+			)?, registry)?,
+			notifications_in: register(CounterVec::new(
+				Opts::new(
+					"sub_libp2p_notifications_in",
```

Comment:
```suggestion
					"sub_libp2p_notifications_in_total",
```

This is an ever increasing counter, right?

Reply:
Yes